eclair-core/src/main/scala/fr/acinq/eclair/channel/Helpers.scala

```diff
@@ -820,9 +820,9 @@ object Helpers {
        * NB: with anchor outputs, it's possible to have transactions that spend *many* HTLC outputs at once, but we're not
        * doing that because it introduces a lot of subtle edge cases.
        */
-      def claimHtlcDelayedOutput(localCommitPublished: LocalCommitPublished, keyManager: ChannelKeyManager, commitment: FullCommitment, tx: Transaction, feeEstimator: FeeEstimator, feeTargets: FeeTargets, finalScriptPubKey: ByteVector)(implicit log: LoggingAdapter): (LocalCommitPublished, Option[HtlcDelayedTx]) = {
+      def claimHtlcDelayedOutput(localCommitPublished: LocalCommitPublished, keyManager: ChannelKeyManager, commitment: FullCommitment, tx: Transaction, onChainFeeConf: OnChainFeeConf, finalScriptPubKey: ByteVector)(implicit log: LoggingAdapter): (LocalCommitPublished, Option[HtlcDelayedTx]) = {
         if (isHtlcSuccess(tx, localCommitPublished) || isHtlcTimeout(tx, localCommitPublished)) {
-          val feeratePerKwDelayed = feeEstimator.getFeeratePerKw(feeTargets.claimMainBlockTarget)
+          val feeratePerKwDelayed = onChainFeeConf.getClosingFeerate()
```

Comment:
I lumped all "non-competitive" cases (mutual close, claim main, claim htlc delayed) in the same `getClosingFeerate()` bucket. I don't see why we should separate them.

Reply:
Agreed, that makes sense! Then we only need the following buckets in `eclair.conf`:

- funding (slow/medium/fast)
- commitment-without-anchors (slow/medium/fast)
- commitment-with-anchors (slow/medium/fast)
- close (slow/medium/fast)

I think we don't need the commitment-without-htlc case anymore, this is handled by separating the anchor/non-anchor cases.

In the anchor case, we will very likely use a defaut of `slow` for the commitment, and the feerate we use when force-closing depends on whether there are HTLCs at risk and when they timeout.

In the non-anchor case, the default should likely be `fast` since we cannot CPFP.

Do you think that makes sense?

Reply:
> Do you think that makes sense?

For funding and close it definitely makes sense to offer configurable priority. But for commitments it should be entirely managed automatically by eclair I think, with a fallback to the regular "close" setting when there are no competing transactions.

Reply:
> People will have different time preference here depending on whether they have big pockets and want to optimize for fees or not so much liquidity and want to move it quickly.

I agree, but why would this setting be different from the `closing` one?

Reply:
> I agree, but why would this setting be different from the closing one?

Probably not, good idea, we can treat this as a closing!